slack_bolt/request/internals.py

```diff
@@ -208,6 +208,26 @@ def extract_channel_id(payload: Dict[str, Any]) -> Optional[str]:
     return None
 
 
+def extract_function_execution_id(payload: Dict[str, Any]) -> Optional[str]:
+    if payload.get("function_execution_id") is not None:
+        return payload.get("function_execution_id")
+    if payload.get("event") is not None:
+        return extract_function_execution_id(payload["event"])
+    if payload.get("function_data") is not None:
+        return extract_function_execution_id(payload["function_data"])
+    if payload.get("execution_id") is not None:
+        return payload.get("execution_id")
+    return None
+
+
+def extract_function_bot_access_token(payload: Dict[str, Any]) -> Optional[str]:
+    if payload.get("bot_access_token") is not None:
```

Comment:
Since this key is not used in any other paterns, this simple implementation is fine so far. If we want to make it safer, checking the existence of either function_execution_id or function_data in the same dict object would be worth considering. That said, this suggestion is totally optional.

Reply:
Good point 💯  I've implemented this in the `build_context` and `build_async_context` functions to reduce unnecessary runtime execution
The `bot_access_token` will only be parsed if there it a `function_execution_id` found in the payload